core/types/tx_pool.go

```diff
@@ -0,0 +1,102 @@
+package types
+
+import (
+	"io"
+	"math/big"
+
+	"github.com/pkg/errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+	staking "github.com/harmony-one/harmony/staking/types"
+)
+
+var (
+	// ErrUnknownPoolTxType is returned when attempting to assert a PoolTransaction to its concrete type
+	ErrUnknownPoolTxType = errors.New("unknown transaction type in tx-pool")
+)
+
+// PoolTransaction is the general transaction interface used by the tx pool
+type PoolTransaction interface {
+	Hash() common.Hash
+	Nonce() uint64
+	ChainID() *big.Int
+	ToShardID() uint32
```

Comment:
where is ShardID() getter?

Reply:
For staking txn, there is no shard for it. But it implicitly means shard 0.

Reply:
yes, though it was used in tx pool, but just check and it does not. Will remove this requirement of the interface and remove the dummy implementation of the method in `StakingTransaction`

Reply:
but don't you still need to check the shardID in txPool and remove those txns that's not for my shard?

Reply:
How is this resolved? (Don't just resolve a comment without a clear closure).

Reply:
the current tx pool doesn't need shardIDs. Will implement this (and possibly Cx as well) in future PR.